stl/inc/valarray

```diff
@@ -1372,6 +1372,11 @@ public:
     _NODISCARD size_t stride() const noexcept /* strengthened */ {
         return _Stride;
     }
+#if _HAS_CXX200
+    friend bool operator==(const slice& x, const slice& y) {

```

Comment:
Another thing we do that goes beyond the Standardese is "strengthening" exception specifications. The Standard permits (but doesn't require) implementations to look at a function and say "our implementation never emits exceptions, so we can mark this as `noexcept` which might improve codegen for users".

In this case, the comparison operator calls only `start()`, `size()`, and `stride()` (and then uses builtin `==` and `&&`). Those member functions are defined immediately above - and as they simply return `size_t` data members, we marked them as `noexcept /* strengthened */`. (We use the comment to indicate that we're providing something beyond the Standard here.) So, this comparison operator should also be marked as `noexcept /* strengthened */` to be consistent.